pkg/jobs/jobclient.go

```diff
@@ -377,79 +377,111 @@ func (c *JobClient) GetLastLogLineError(ctx context.Context, podNamespace, podNa
 }
 
 // GetPodLogs returns pod logs bytes
-func (c *JobClient) GetPodLogs(podName string, logLinesCount ...int64) (logs []byte, err error) {
+func (c *JobClient) GetPodLogs(pod corev1.Pod, logLinesCount ...int64) (logs []byte, err error) {
 	count := int64(100)
 	if len(logLinesCount) > 0 {
 		count = logLinesCount[0]
 	}
 
-	podLogOptions := corev1.PodLogOptions{
-		Follow:    false,
-		TailLines: &count,
+	var containers []string
+	for _, container := range pod.Spec.InitContainers {
+		containers = append(containers, container.Name)
 	}
 
-	podLogRequest := c.ClientSet.CoreV1().
-		Pods(c.Namespace).
-		GetLogs(podName, &podLogOptions)
-
-	stream, err := podLogRequest.Stream(context.TODO())
-	if err != nil {
-		return logs, err
+	for _, container := range pod.Spec.Containers {
+		containers = append(containers, container.Name)
 	}
 
-	defer stream.Close()
+	for _, container := range containers {
+		podLogOptions := corev1.PodLogOptions{
+			Follow:    false,
+			TailLines: &count,
+			Container: container,
+		}
 
-	buf := new(bytes.Buffer)
-	_, err = io.Copy(buf, stream)
-	if err != nil {
-		return logs, err
+		podLogRequest := c.ClientSet.CoreV1().
+			Pods(c.Namespace).
+			GetLogs(pod.Name, &podLogOptions)
+
+		stream, err := podLogRequest.Stream(context.TODO())
+		if err != nil {
+			if len(logs) != 0 {
+				return logs, nil
+			}
+
+			return logs, err
+		}
+
+		defer stream.Close()
+
+		buf := new(bytes.Buffer)
+		_, err = io.Copy(buf, stream)
+		if err != nil {
+			if len(logs) != 0 {
+				return logs, nil
```

Comment:
same here, should we make sure it's the right type of error before returning half-streamed logs with no error?

Reply:
same approach

Reply:
https://github.com/kubernetes/test-infra/blob/d408ead6e01bf53681bb1758f5a4728b28628fdd/prow/cmd/deck/main.go#L1368